### ISSUE TITLE
Can't add new to interceptor

### ISSUE THREAD

**Issue by ASDAlexey** [dscheerens/ngx-forkable-http-client#5]
I've tryed to use this cool lib, but and could not figure out how to add providers to use from within the interceptor?

For example:
to app.module I've added 

providers:[
...
// Define the providers for the non-global interceptors.
    JwtInterceptor,

    // Define the HTTP clients.
    {
      provide: JWT,
      useFactory: forkHttpClient,
      deps: [ForkableHttpClient, JwtInterceptor],
    },
]

jwt.interseptor.ts
@Injectable({ providedIn: 'root' })
export class JwtInterceptor implements HttpInterceptor {
  constructor(private authService: AuthService) {}

intercept(req: HttpRequest<any>, next: HttpHandler): any {
    debugger;
   // this.authService - IS EMPTY HERE
    const tokens: Tokens = this.authService.getTokens();
}

How can I add any providers to interceptor constructor?

**Comment by dscheerens** [comment]
Hmm, that's strange. Can't see anything wrong with this code :confused: 

Just to check: your problem is that at runtime `authService` is `undefined` inside the `JwtInterceptor.intercept` function?

Also, do you get any any errors on application startup about the Angular not being able to resolve all providers for `JwtInterceptor`?

**Comment by ASDAlexey** [comment]
Yes problem is that at runtime authService is undefined inside the JwtInterceptor.intercept function.
In case if I use global interceptor user service is defened
https://monosnap.com/file/fRQNXuWSBrrNWVbgTIRHhdE6ZBei7m
but in case if I use ngx-forkable-http-client any services in jWTInteceptor is empty 
https://monosnap.com/file/oXxU53Yth58454renFOjRymSQktXNg

**Comment by ASDAlexey** [comment]
application haven't got any error in startup about Angular

**Comment by ASDAlexey** [comment]
Thank you for help. I've find the problem. https://monosnap.com/file/St6KhphglFYqfbPAz4WcLuXVIYrPVu
The problem was that I tried to connect inside authService @Inject (JWT) private httpJWT: HttpClient 
and inside JWTInterceptor tried to connect authService https://monosnap.com/file/tgvMlKiChil8Lp4Glsv8BlP1Jy7Ong
After I stopped using @Inject (JWT) private httpJWT: HttpClient in authService, the problem disappeared

Thank you for your help. I will use this wonderful library thank you. I've added start to this repo.
